src/delay.rs

```diff
@@ -31,11 +33,17 @@ impl Delay {
         let ticks_per_us = freq / 1_000_000_u32;
         Delay { ticks_per_us, syst }
     }
+
+    /// Wait for the given time.
+    ///
+    /// Note that durations above `u32::MAX` microseconds will be clamped at `u32::MAX`.
```

Comment:
I'm not particularly happy with this behaviour but I can't think of a better solution right now. Thoughts appreciated!

Reply:
I'm not happy about that either, but can't think of a better solution right now. A fallible interface isn't great, panicking isn't great either. Your solution here at least should work for the common case, and the pitfall is documented.